### DISCUSSION TITLE
get current router in middleware

### DISCUSSION THREAD

**Discussion by tingfeng-key** [tokio-rs/axum#1058] [Q&A] [answered]
How to get the current access route in middleware or handler? I want to use this to realize the restrictions of different user identities on access routes

**Comment by davidpdrsn** [comment, marked as answer]
You can extract either [`Uri`](https://docs.rs/http/latest/http/uri/struct.Uri.html):

```rust
async fn handle(uri: axum::http::Uri) {}
```

Or [`MatchedPath`](https://docs.rs/axum/latest/axum/extract/struct.MatchedPath.html):

```rust
async fn handle(uri: axum::extract::MatchedPath) {}
```

**Reply by tingfeng-key** [reply]
Thank you for your reply. Is there any way to get all the routes currently defined?

**Reply by davidpdrsn** [reply]
Nope. You'd have to build your own thing for that. You could make your own router that wraps `axum::Router` and tracks the routes that are added.

**Reply by tingfeng-key** [reply]
OK, thank you
